src/stream/src/executor/source/iceberg_fetch_executor.rs

```diff
@@ -0,0 +1,451 @@
+                            for chunk in chunks {
+                                println!(
+                                    "chunk:\n {:}, file_path_idx: {}, file_pos_idx: {}",
+                                    chunk.to_pretty(),
+                                    file_path_idx,
+                                    file_pos_idx
+                                );
+                                // let mapping =
+                                //     get_split_offset_mapping_from_chunk(&chunk, split_idx, offset_idx)
+                                //         .unwrap();
+                                let mut mapping = HashMap::new();
+                                // All rows (including those visible or invisible) will be used to update the source offset.
+                                // TODO: maybe we can get offset separately, instead of in the chunk.
+                                for i in 0..chunk.capacity() {
+                                    let (_, row, _) = chunk.row_at(i);
+                                    let file_path: Arc<str> =
+                                        row.datum_at(file_path_idx).unwrap().into_utf8().into();
+                                    let file_pos = row.datum_at(file_pos_idx).unwrap().into_int64();
+                                    mapping.insert(file_path, file_pos.to_owned());
+                                }
+                                debug_assert_eq!(mapping.len(), 1);
+                                if let Some((file_path, _file_pos)) = mapping.into_iter().next() {
+                                    let row = state_store_handler
+                                        .get(&file_path)
+                                        .await?
+                                        .expect("The fs_split should be in the state table.");
+                                    let _fs_split: FileScanTask = match row.datum_at(1) {
+                                        Some(ScalarRefImpl::Jsonb(jsonb_ref)) => {
+                                            serde_json::from_value(
+                                                jsonb_ref.to_owned_scalar().take(),
+                                            )
+                                            .with_context(|| {
+                                                format!("invalid state: {:?}", jsonb_ref)
+                                            })?
+                                        }
+                                        _ => unreachable!(),
+                                    };
+
+                                    // TODO: support persist progress after supporting reading part of a file.
+                                    if true {
+                                        splits_on_fetch -= 1;
+                                        state_store_handler.delete(&file_path).await?;
+                                    }
+                                }
```

Comment:
I don't quite get this part. Could you please illustrate how it works?

Reply:
This was simply copied from `FsFetchExecutor`. I forgot to examine and update it  🤡